server/plugin/command.go

```diff
@@ -453,6 +459,151 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
 	}
 }
 
+func (p *Plugin) handlewebhook(_ *plugin.Context, args *model.CommandArgs, parameters []string, userInfo *GitHubUserInfo) string {
+	if len(parameters) == 0 {
+		return "Invalid webhook command. Available command are `add` and `list` ."
+	}
+	command := parameters[0]
+	parameters = parameters[1:]
+	ctx := context.Background()
+	githubClient := p.getGithubClient(userInfo)
+	baseURL := p.getBaseURL()
+	switch command {
+	case add:
+		if len(parameters) < 1 {
+			return "Unknown action, Currently supported to add and List webhook "
+		}
+		owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+		if owner == "" {
+			return "Currently supported to add webhook to a specific [owner/repo] ."
+		}
+
+		siteURL := *p.API.GetConfig().ServiceSettings.SiteURL + inboundWebhookURL
+		var hook github.Hook
+		var config = make(map[string]interface{})
+
+		if len(parameters) > 2 {
+			for _, val := range strings.Split(parameters[2], ",") {
+				KVPair := strings.Split(val, "=")
+				switch {
+				case KVPair[0] == "insecure_ssl":
+					insecureSSL, err := strconv.ParseBool(KVPair[1])
+					if err != nil {
+						return "config filed insecure_ssl can only have on of the following true or false "
+					}
+					config[KVPair[0]] = insecureSSL
+				case KVPair[0] == "options":
+					hook.Events = strings.Split(KVPair[1], "&")
+				default:
+					config[KVPair[0]] = KVPair[1]
+				}
+			}
+		} else {
+			config["url"] = siteURL
+			config["insecure_ssl"] = false
+			config["content_type"] = "application/json"
```

Comment:
The GitHub UI defaults to urlencoded. Should we do the same?

Reply:
much better to keep default values same as in github

Reply:
@mickmister can you check this , that seems to be a outdated code [here](https://github.com/mattermost/mattermost-plugin-github/pull/456/files#r739644685)